components/script/dom/gpubuffer.rs

```diff
@@ -256,10 +255,72 @@ impl GPUBufferMethods for GPUBuffer {
 
         self.state.set(GPUBufferState::MappingPending);
         self.map_mode.set(Some(mode));
-        *self.mapping_range.borrow_mut() = map_range;
+        *self.mapping_range.borrow_mut() = Some(map_range);
+        *self.map_promise.borrow_mut() = Some(promise.clone());
         promise
     }
 
+    /// https://gpuweb.github.io/gpuweb/#dom-gpubuffer-getmappedrange
+    #[allow(unsafe_code)]
+    fn GetMappedRange(
+        &self,
+        cx: JSContext,
+        offset: GPUSize64,
+        size: GPUSize64,
+    ) -> Fallible<NonNull<JSObject>> {
+        if self.mapped_ranges.borrow().is_none() {
```

Comment:
It seems strange to have this lazy check here. We should know exactly when this function is valid to call (either after mapping, or in a buffer that's created with mapping), so those earlier stages need to prepare the state, and code here doesn't need to be lazy